### PR TITLE
Add createJestMockInstance

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:
[No issue]

Attempts to emulate sinon's [createStubInstance](https://sinonjs.org/releases/latest/stubs/) API for `jest`. The main motivation was being able to easily create type-safe mocks, but it will be also handy in case we want to move from `mocha` to `jest` in existing repos

The function signature is slightly different since the mocking logic is different between the two frameworks.

### CODE REVIEW

**Review by edmofro — APPROVED** [review]
Love it!

**Comment by kael89** [comment]
Fixed failing build in the last commit

File: packages/utils/src/testUtilities/createJestMockInstance.js [hunk 1]
```diff
@@ -3,7 +3,6 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { jest } from '@jest/globals';
```
Comment (edmofro) [inline packages/utils/src/testUtilities/createJestMockInstance.js]: So where does it get `jest` from below?

Comment (kael89) [reply packages/utils/src/testUtilities/createJestMockInstance.js]: It's global during tests! Same with `expect`, `describe` etc

Comment (kael89) [reply packages/utils/src/testUtilities/createJestMockInstance.js]: I knew that but I thought explicitly importing `jest` would be a better practice. But `jest` complains, it seems you can only do that in test files, which kind of beats the purpose 🤔

Comment (kael89) [reply packages/utils/src/testUtilities/createJestMockInstance.js]: I guess they want to discourage people from loading `jest` from `@jest/globals` in a non test environment and expect it'll work (it won't)

**Review by edmofro — APPROVED** [review]

**Pull request merged**